src/ScriptEditor/ui/ScriptEditorRoot.tsx

```diff
@@ -143,6 +143,7 @@ function Root(props: IProps): React.ReactElement {
     infLoop(newCode);
     updateRAM(
       !currentScript || currentScript.isTxt ? null : newCode,
+      currentScript?.path ?? null,
```

Comment:
`?? null` should be unneeded here?

Reply:
`if (newCode === null || filename === null || server === null) {`
it explicity looks for null in updateRAM thats why i assumed it has to be null not undefined
and i tried to keep the changes (for now/this pr) in that section to as little as possible

Reply:
But currentScript can't be undefined? 

If you're worried about it, switch to `== null` to catch both nullish values.

Reply:
its intialized with null thats why the other two arguments check for truthyness of currentscript there
its actually a hazard because we have like a dozen of null/truthyness checks for it because of that

i would like to change that when as a whole but assumed thats out of scope of this pr

Reply:
Ah, I thought optional chaining passed the value through (i.e. `(null)?.foo === null`), but I see that was wrong. But regardless, you can handle it by changing the check in the function to `== null` (either for just this term or all of them),  which catches null and undefined both.

Reply:
i would go with 
`currentScript && currentScript.path,`
when thats okay cuz then it fits style wise to the other check
i can also change the === null's to == null

Reply:
`currentScript && currentScript.path` is the same as `currentScript?.path` but less clear/more bug prone IMO. For instance it would be 0 if currentScript were 0; that shouldn't happen, but that's why nullish coalescing exists.